test/xpu_api/numerics/numeric.ops/numeric.ops.gcd/xpu_gcd.pass.cpp

```diff
@@ -0,0 +1,135 @@
+// -*- C++ -*-
+//===-- xpu_gcd.pass.cpp --------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/numeric>
+#include <oneapi/dpl/type_traits>
+
+#include <iostream>
+#include <CL/sycl.hpp>
+
+constexpr sycl::access::mode sycl_read = sycl::access::mode::read;
+constexpr sycl::access::mode sycl_write = sycl::access::mode::write;
+
+template <typename T1, typename T2>
+class KernelName;
+
+template <typename Input1, typename Input2, typename Output>
+bool
+test0(int in1, int in2, int out)
+{
+    auto value1 = static_cast<Input1>(in1);
+    auto value2 = static_cast<Input2>(in2);
+    static_assert(oneapi::dpl::is_same<Output, decltype(oneapi::dpl::gcd(value1, value2))>::value, "");
+    static_assert(oneapi::dpl::is_same<Output, decltype(oneapi::dpl::gcd(value2, value1))>::value, "");
+    return (static_cast<Output>(out) == oneapi::dpl::gcd(value1, value2));
+}
+
+template <typename Input1, typename Input2 = Input1>
+void
+do_test()
+{
+    sycl::queue deviceQueue;
+    bool res = true;
+    sycl::range<1> numOfItems1{1};
+
+    {
+        sycl::buffer<bool, 1> buffer1(&res, numOfItems1);
+        deviceQueue.submit([&](sycl::handler& cgh) {
+            auto out = buffer1.get_access<sycl_write>(cgh);
+            cgh.single_task<KernelName<Input1, Input2>>([=]() {
+                using S1 = oneapi::dpl::make_signed_t<Input1>;
+                using S2 = oneapi::dpl::make_signed_t<Input2>;
+                using U1 = oneapi::dpl::make_unsigned_t<Input1>;
+                using U2 = oneapi::dpl::make_unsigned_t<Input2>;
+                constexpr struct
+                {
+                    int x;
+                    int y;
+                    int expect;
+                } Cases[] = {{0, 0, 0}, {1, 0, 1}, {0, 1, 1},   {1, 1, 1},
+                             {2, 3, 1}, {2, 4, 2}, {36, 17, 1}, {36, 18, 18}};
+                for (auto TC : Cases)
```

Comment:
it's uncommon to use capitals or names starting from capitals as the variable names in C++. I would suggest two rename both to `tc` and `cases` but I don't insist. If you decide to do that please modify another file as well.